etna/collections/models.py

```diff
@@ -400,16 +413,29 @@ def get_datalayer_data(self, request: HttpRequest) -> Dict[str, Any]:
 
     @cached_property
     def related_articles(self):
-        from etna.articles.models import ArticlePage
+        """Return a list of related pages for rendering in the related articles section
+        of the page. To add another page type, import it and add it to the related_page_types list.
+        """
 
-        return (
-            ArticlePage.objects.exclude(pk=self.featured_article)
-            .live()
-            .public()
-            .filter(pk__in=self.related_page_pks)
-            .order_by("-first_published_at")
-            .select_related("teaser_image")
-        )
+        from etna.articles.models import ArticlePage, FocusedArticlePage
+
+        related_page_types = [ArticlePage, FocusedArticlePage]
+        page_list = []
+
+        for page_type in related_page_types:
+            pages = (
+                page_type.objects.exclude(pk=self.featured_article_id)
+                .filter(pk__in=self.related_page_pks)
+                .live()
+                .public()
+                .select_related("teaser_image")
+            )
+            if pages:
```

Comment:
The `if` here will actually be introducing an additional `exists()` query here, which we don't want. `page_list.extend(pages)` will happily handle an empty result, so the conditional can be ditched.